repotracker/repotracker.go

```diff
@@ -693,130 +705,127 @@ func sanityCheckOrderNum(revOrderNum int, projectId, revision string) error {
 
 // createVersionItems populates and stores all the tasks and builds for a version according to
 // the given project config.
-func createVersionItems(v *model.Version, ref *model.ProjectRef, metadata VersionMetadata, project *model.Project) error {
-	// generate all task Ids so that we can easily reference them for dependencies
-	sourceRev := ""
-	if metadata.SourceVersion != nil {
-		sourceRev = metadata.SourceVersion.Revision
-	}
-	var aliases model.ProjectAliases
-	var err error
-	if metadata.Alias != "" {
-		aliases, err = model.FindAliasInProject(ref.Identifier, metadata.Alias)
-		if err != nil {
-			return errors.Wrap(err, "error finding project alias")
+func createVersionItems(ctx context.Context, v *model.Version, ref *model.ProjectRef, metadata VersionMetadata, project *model.Project, aliases model.ProjectAliases) error {
+	client := evergreen.GetEnvironment().Client()
+
+	return client.UseSession(ctx, func(sessCtx mongo.SessionContext) error {
+		// generate all task Ids so that we can easily reference them for dependencies
+		sourceRev := ""
+		if metadata.SourceVersion != nil {
+			sourceRev = metadata.SourceVersion.Revision
 		}
-	}
-	taskIds := model.NewTaskIdTable(project, v, sourceRev, metadata.DefinitionID)
+		taskIds := model.NewTaskIdTable(project, v, sourceRev, metadata.DefinitionID)
 
-	// create all builds for the version
-	for _, buildvariant := range project.BuildVariants {
-		if buildvariant.Disabled {
-			continue
+		err := sessCtx.StartTransaction()
+		if err != nil {
+			return errors.Wrap(err, "error starting transaction")
 		}
-		var match bool
-		if len(aliases) > 0 {
-			match, err = aliases.HasMatchingVariant(buildvariant.Name)
-			if err != nil {
-				grip.Error(err)
+		// create all builds for the version
+		for _, buildvariant := range project.BuildVariants {
+			if buildvariant.Disabled {
 				continue
 			}
-			if !match {
-				continue
+			var match bool
+			if len(aliases) > 0 {
+				match, err = aliases.HasMatchingVariant(buildvariant.Name)
+				if err != nil {
+					grip.Error(err)
+					continue
+				}
+				if !match {
+					continue
+				}
 			}
-		}
-		args := model.BuildCreateArgs{
-			Project:      *project,
-			Version:      *v,
-			TaskIDs:      taskIds,
-			BuildName:    buildvariant.Name,
-			Activated:    false,
-			SourceRev:    sourceRev,
-			DefinitionID: metadata.DefinitionID,
-			Aliases:      aliases,
-		}
-		var buildId string
-		buildId, err = model.CreateBuildFromVersion(args)
-		if err != nil {
-			if metadata.TriggerID == "" || !strings.Contains(err.Error(), "E11000") {
```

Comment:
What's the reason for removing the `metadata.TriggerID` check?

Reply:
This entire block isn't needed anymore because it was part of an attempt to continue on error if we error inserting some documents. We abort the transaction now